doc/Doxyfile-doxygen

```diff
@@ -0,0 +1,2432 @@
+# Doxyfile 1.8.13
```

Comment:
Its really hard to review this. My initial question was: Why is this needed? I guess its to output the info to xml? Turns out its the other way around now. Anyways I would do it this way:
1. leave the normal doxygen file for doxygen
2. then change this filename to e.g. Doxyfile-rtd and use `@INCLUDE = config_file_name` and just overwrite values (see http://www.doxygen.nl/manual/config.html). So just use the normal doxyfile as a base and just modify a few lines that you want to change.

This makes it really easy to understand and keeps things synchronized (we would like both docs to show the highest degree of identical content, right).

Reply:
I don't like it either. Unfortunately though, Exhale requires a file named `Doxyfile` and its settings need to be slightly different than the original `Doxyfile` (for instance, because `exhale` uses XML, not HTML output). This is why there are two versions of the Doxyfile now.

I didn't know about the Include thing, I'll look into it.

Reply:
https://exhale.readthedocs.io/en/latest/reference/configs.html#exhale.configs.exhaleDoxygenStdin